templates/ecs.yaml

```diff
@@ -0,0 +1,317 @@
+AWSTemplateFormatVersion: '2010-09-09'
+Parameters:
+  KeyName:
+    Type: AWS::EC2::KeyPair::KeyName
+    Description: Name of an existing EC2 KeyPair to enable SSH access to the ECS instances.
+  VpcId:
+    Type: AWS::EC2::VPC::Id
+    Description: Select a VPC that allows instances access to the Internet.
+  BastionHostSgId:
+    Type: String
+    Description: The ID of the security group for bastion hosts
+  PublicSubnetId:
+    Type: List<AWS::EC2::Subnet::Id>
+    Description: Public subnet IDs from the VPC for ALB placement. Minimum of two.
+  PrivateSubnetId:
+    Type: List<AWS::EC2::Subnet::Id>
+    Description: Private subnet IDs from the VPC for ECS node placement. Minimum of two.
+  DesiredCapacity:
+    Type: Number
+    Default: '1'
+    Description: Number of instances to launch in your ECS cluster.
+  MaxSize:
+    Type: Number
+    Default: '1'
+    Description: Maximum number of instances that can be launched in your ECS cluster.
+  InstanceType:
+    Description: EC2 instance type
+    Type: String
+    Default: t3a.micro
+    AllowedValues: [t3.micro, t3.small, t3.medium, t3.large, t3a.micro, t3a.small, t3a.medium, t3a.large]
+    ConstraintDescription: Please choose a valid instance type.
+  DockerImageTag:
+    Type: String
+    Description: The Docker image tae to run
+Mappings:
+  AWSRegionToAMI:
+    us-east-1:
+      AMIID: ami-0669eafef622afea1
+    us-west-2:
+      AMIID: ami-05d33de432484ef34
+Resources:
+  # iAtlas API task execution IAM role
+  # This is necessary so it can read the Secrets Manager entry that contains the external
+  # registry credentials
+  IAtlasIAMExecutionRole:
+    Type: AWS::IAM::Role
+    Properties:
+      AssumeRolePolicyDocument:
+        Version: '2012-10-17'
+        Statement:
+          - Effect: Allow
+            Principal:
+              Service:
+                - ecs-tasks.amazonaws.com
+            Action:
+              - 'sts:AssumeRole'
+      Path: /
+      Policies:
```

Comment:
an AWS best practice is to add managed policies to the role instead of adding inline policies.  take a look at https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-iam-role.html#cfn-iam-role-managepolicyarns